### PR TITLE
support unicode characters in string literals up to `0.7.0`

### LINKED ISSUE #784: string literals should accept unicode characters before `0.7.0`
The following string literal is legal until `0.7.0`. Afterwards, it became disallowed after the introduction of `UnicodeStringLiteral`:

```solidity
contract Test {
    event Bar(string y);

    function foo() public {
        emit Bar("Fireworks 🎆");
    }
}
```

### PR DESCRIPTION
**Pull request by OmarTawfik**
Closes #784

### CODE REVIEW

File: crates/solidity/inputs/language/src/definition.rs [hunk 1]
```diff
@@ -3700,26 +3700,24 @@ codegen_language_macros::compile!(Language(
                         Token(
                             name = HexStringLiteral,
                             definitions = [
-                                TokenDefinition(scanner = Fragment(SingleQuotedHexStringLiteral)),
```
[the remark this answers is not in the document]
Comment (OmarTawfik) [reply crates/solidity/inputs/language/src/definition.rs:3703]: Discussed face to face, let's separate them into distinct tokens.

File: crates/solidity/inputs/language/src/definition.rs [hunk 2]
```diff
@@ -3744,36 +3742,58 @@ codegen_language_macros::compile!(Language(
                         Token(
                             name = AsciiStringLiteral,
```
[the remark this answers is not in the document]
Comment (OmarTawfik) [reply crates/solidity/inputs/language/src/definition.rs:3743]: I like this. Let me try it out.

**Comment by OmarTawfik** [comment]
@Xanewok comments addressed. Please let me know if you have any other suggestions.

**Comment by Xanewok** [comment]
LGTM as well but needs rebase

**Comment by OmarTawfik** [comment]
@Xanewok needs a ✅  stamp still.

**Review by Xanewok — APPROVED** [review]

**Pull request merged**